### PR TITLE
Added NewlineAfterOpenTagFixer

### PR DESCRIPTION
**Pull request by ceeram**
Adding new fixer: NewlineAfterOpenTag. refs #641

### CODE REVIEW

File: Symfony/CS/Fixer/Symfony/NewlineAfterOpenTagFixer.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            }
+
+            if (false !== strpos($token->getContent(), "\n")) {
+                break;
+            }
+
+            $newlines = 0;
+            $whitespaceTokens = $tokens->findGivenKind(T_WHITESPACE);
+            foreach ($whitespaceTokens as $whitespaceToken) {
+                !$whitespaceToken->isWhitespace(array('whitespaces' => "\n")) ?: ++$newlines;
```
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/NewlineAfterOpenTagFixer.php]: No readability IMO. `?:` is not for that

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+     * @dataProvider provideCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideCases()
+    {
+        return array(
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: please add tests with multiple `<?php` and `<?=`

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+$foo = true;
+?>',
+            ),
+            array(
+                '<?php
+$foo = true;
+$bar = false;
+?>',
+                '<?php $foo = true;
+$bar = false;
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: plase try sth like
```php
    <?php $a = 1;
    $b = 3;
```
into
```php
    <?php
    $a = 1;
    $b = 3;
```

**Comment by keradus** [comment]
Would you add a second fixer that is requested in #641 , that fix
```php
 <?php namespace Vendor\Namespace;
```
into
```php
<?php

namespace Vendor\Namespace;
```
please?

File: Symfony/CS/Fixer/Symfony/NewlineAfterOpenTagFixer.php (hunk 1 above)
Comment (ceeram) [inline Symfony/CS/Fixer/Symfony/NewlineAfterOpenTagFixer.php]: It saved me a level of nesting, ill change it

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php (hunk 3 above)
Comment (ceeram) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: how is that different then the code i have? its exactly doing that isnt it?

**Comment by ceeram** [comment]
Sure can do, i suppose that would be Contrib level then, as it would conflict with the fixer of this PR

**Comment by ceeram** [comment]
See also #1032

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: it prove that `$a = 1;` line is moved keeping indent intact.

**Comment by keradus** [comment]
> Sure can do, i suppose that would be Contrib level then, as it would conflict with the fixer of this PR

why it would be a conflict?
this fixer add enter if it is missing, next fixer add second enter to have one line free

**Comment by keradus** [comment]
Oh, you create fixer to add blank line after `<?php`, not before `namespace ...`. Cool.

If this fixer (NewlineAfterOpenTag) change:
```php
<?php


$a = 1;
```

into:
```
<?php
$a = 1;
```


if not - they are not conflicting.
If yes - then `BlanklineAfterOpenTagFixer` should be on symfony level, not this one.

**Comment by ceeram** [comment]
Right, they are not conflicting indeed, but doesnt make sense to run both, as the other one, does what this one does, plus a bit more

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php (hunk 3 above)
Comment (ceeram) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: Ah ok, missed the indent

Comment (ceeram) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: BTW, you have whitespace before open tag as well

**Comment by keradus** [comment]
then please set `BlanklineAfterOpenTagFixer` at Symfony and this at Contrib level.

We plan to handle that cases on 2.0 - to let one fixer turn off the other one. List created here: #1033

**Comment by ceeram** [comment]
OK, switched the levels of both

File: Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: yes, if open tag wasn't indented there would be no point to indent next line

Comment (ceeram) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: i wonder if its really wise to fix the indented code as well, as this is not a php only file, but mixing html with php. I think most people mixing html and php dont want the newline introduced?

Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: If a file contains both php and html then it is not a monolithic file and this fixer will ignore that file, is it not ?

it is some strange case of
```php
   <?php $foo = 1;
        $bar = 2;
```

Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/NewlineAfterOpenTagFixerTest.php]: ok, I see your point. `    <?php` is already html+php ;)

**Comment by keradus** [comment]
Merged into #1089, thank you @ceeram !

**Pull request closed without merging**